src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -82,13 +81,31 @@ public List<SourceRecord> poll() throws InterruptedException {
         return newSourceRecords.isEmpty() ? null : newSourceRecords;
     }
 
-    protected RowBatcher<JsonNode> newRowBatcher(List<SourceRecord> newSourceRecords) {
-        ContentHandle<JsonNode> jsonHandle = new JacksonHandle().withFormat(Format.JSON).withMimetype("application/json");
-        rowBatcher =  dataMovementManager.newRowBatcher(jsonHandle);
-        configureRowBatcher(parsedConfig, rowBatcher, newSourceRecords);
+    protected RowBatcher<?> newRowBatcher(List<SourceRecord> newSourceRecords) {
+        rowBatcher = rowBatcherBuilder.newRowBatcher(newSourceRecords);
+        configureRowBatcher(parsedConfig, rowBatcher);
         return rowBatcher;
     }
 
+    protected RowBatcherBuilder<?> newRowBatcherBuilder(DataMovementManager dataMovementManager, Map<String, Object> parsedConfig) {
```

Comment:
Make this and configureRowBatcher private - the benefit being that it makes it more obvious which methods are exposed for testing (the protected ones).